integrations/Iterable/browser.js

```diff
@@ -0,0 +1,211 @@
+            isRequiredToDismissMessage: this.isRequiredToDismissMessage || undefined,
+            position: this.closeButtonPosition || 'top-right'
+          }
+        }
+    getInAppMessagesPayload = removeUndefinedAndNullValues(getInAppMessagesPayload);
+
+    const { request } = window['@iterable/web-sdk'].getInAppMessages(
+        getInAppMessagesPayload,
+        { display: 'immediate' }
+        );
+    // fetchAppEvents is a class function now available throughout
+    // we will trigger getInAppMessages when event name matches from the ui mapping in config.
+    this.fetchAppEvents = request;
+  }
+
+  track(rudderElement) {
+    logger.debug("===In track Iterable===");
+    
+    const { message } = rudderElement;
+    const { event } = message;
+    const eventPayload = removeUndefinedAndNullValues(message.properties);
+    const userEmail = get(message, "context.traits.email");
+    const userId = get(message, "userId");
+    
+    if (!event) {
+        logger.error("Event name not present");
+        return;
+    }
+    let isCustom = true;
+    if (isNotEmpty(this.getInAppEventMapping)) {
+        const mappedEvents = this.getInAppEventMapping;
+        mappedEvents.forEach((e) => {
+            if(e.eventName === event) {
+                // send a track call for getinappMessages if option enabled in config
+                if (this.sendTrackForInapp) {
+                    window['@iterable/web-sdk'].track({ email: userEmail, userId, eventName: "Track getInAppMessages", dataFields: eventPayload })
+                    .then(logger.debug("Track a getinappMessages event."));
+                }
+                this.fetchAppEvents();
+                isCustom = false;
+            }
+        })
+    }
+    if (isNotEmpty(this.purchaseEventMapping)) {
+        // purchase events
+        const mappedEvents = this.purchaseEventMapping;
+        mappedEvents.forEach((e) => {
+            if(e.eventName === event) {
+                const purchaseEventPayload = formPurchaseEventPayload(message)
+                window['@iterable/web-sdk'].trackPurchase(
+                purchaseEventPayload,
+                )
+                isCustom = false;
+            }
+        })
+    }
+    if (isCustom) {
+        // custom events if event is not mapped
+        /* fields available for custom track event
+        {
+            "email": "string",
+            "userId": "string",
+            "eventName": "string",
+            "id": "string",
+            "createdAt": 0,
+            "dataFields": {},
+            "campaignId": 0,
+            "templateId": 0
+        }
+        */
+        // Either email or userId must be passed in to identify the user.
+        // If both are passed in, email takes precedence.
+        logger.debug(`The event ${event} is not mapped in the dashboard, firing a custom event`);
+        window['@iterable/web-sdk'].track({ email: userEmail, userId, eventName: event, dataFields: eventPayload })
```

Comment:
there is no key for `userId` we may have missed it?

Reply:
userId essentially translates to 
userId : userId which is declared at the top of track invocation.

Reply:
oh ok